realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -591,7 +591,8 @@ static boolean compactRealm(final RealmConfiguration configuration) {
     /**
      * Migrates the Realm file defined by the given configuration using the provided migration block.
      *
-     * @param configuration configuration for the Realm that should be migrated.
+     * @param configuration configuration for the Realm that should be migrated. If this is a SyncConfiguration this
+     *                      method does nothing.
```

Comment:
I think it's better to throw an `UnsupportedOperationException` to prevent the user from providing a migration block for a SyncConfiguration